cellar_rebalancer/src/collector/poller.rs

```diff
@@ -117,10 +212,78 @@ impl<T: 'static + Middleware> Poller<T> {
         }
 
         if std::env::var("CELLAR_DRY_RUN").expect("Expect CELLAR_DRY_RUN var") == "TRUE" {
-            Ok(())
+            ()
         } else {
             tick_info.reverse();
-            self.contract_state.rebalance(tick_info).await
+            let delegate_cosmos_address =
+                self.cosmos_key.to_address(&contact.get_prefix()).unwrap();
+            let name = &config.keys.rebalancer_key;
+            let cosmos_key = config.load_deep_space_key(name.clone());
+
+            let fee = Coin {
+                denom: gas_price.1,
+                amount: 0u32.into(),
+            };
+
+            match timeout(Duration::from_secs(100), async {
+                loop {
+                    // Waiting for new vote period
+                    let res = somm_send::query_commit_period(grpc_client).await;
+                    if let Ok(val) = res {
+                        if val.vote_period_start == val.current_height {
+                            break val;
+                        }
+                    }
+                    sleep(Duration::from_secs(1)).await;
+                }
+            })
+            .await
+            {
+                Ok(val) => {
+                    // Sending Pre-commits
+                    somm_send::send_precommit(
+                        contact,
+                        delegate_cosmos_address,
+                        cosmos_key,
+                        fee.clone(),
+                        vec![self.allocation_precommit().await],
+                    )
+                    .await
+                    .unwrap();
+                }
+                Err(e) => {
+                    println!("Couldn't Send Precommit {:?}", e);
+                }
+            }
+
+            match timeout(Duration::from_secs(100), async {
+                loop {
+                    // Checking Pre-commits for validators
+                    let res = somm_send::query_allocation_precommits(grpc_client).await;
+                    if let Ok(val) = res {
+                        break val;
```

Comment:
Ideally we delay for at least a block here before breaking and proceeding to commit, but this isn't a blocker for merge.